src/tracing.rs

```diff
@@ -67,6 +46,362 @@ pub fn setup(user_filters: &str) -> anyhow::Result<()> {
     ));
 
     tracing::subscriber::set_global_default(reg)?;
+    tracing_log::LogTracer::init()?;
+
+    Ok(logger_handle)
+}
+
+#[derive(Clone)]
+pub struct LoggerHandle {
+    config: Arc<RwLock<config::LoggerConfig>>,
+    default: DefaultLoggerReloadHandle,
+}
+
+#[rustfmt::skip] // `rustfmt` formats this into unreadable single line :/
+type DefaultLoggerReloadHandle<S = Registry> = reload::Handle<
+    default::Logger<S>,
+    S,
+>;
+
+impl LoggerHandle {
+    pub fn new(config: config::LoggerConfig, default: DefaultLoggerReloadHandle) -> Self {
+        Self {
+            config: Arc::new(RwLock::new(config)),
+            default,
+        }
+    }
+
+    pub async fn get_config(&self) -> config::LoggerConfig {
+        self.config.read().await.clone()
+    }
+
+    pub async fn update_config(&self, diff: config::LoggerConfigDiff) -> anyhow::Result<()> {
+        let mut config = self.config.write().await;
+
+        self.default
+            .modify(|logger| default::update(logger, &diff.default))?;
+
+        config.default.update(diff.default);
+
+        Ok(())
+    }
+}
+
+pub mod config {
+    use super::*;
+
+    #[derive(Clone, Debug, Default, Deserialize, Serialize)]
+    pub struct LoggerConfig {
+        #[serde(flatten)]
+        pub default: default::Config,
+    }
+
+    impl LoggerConfig {
+        pub fn with_top_level_directive(&mut self, log_level: Option<String>) -> &mut Self {
+            if self.default.log_level.is_some() && log_level.is_some() {
+                // TODO: Warn if both top-level `log_level` and `LoggerConfig::log_level` directives are used
+                eprintln!("TODO");
+            }
+
+            self.default.log_level = self.default.log_level.take().or(log_level);
+            self
+        }
+
+        pub fn update(&mut self, diff: LoggerConfigDiff) {
+            self.default.update(diff.default);
+        }
+    }
+
+    #[derive(Clone, Debug, Default, Deserialize)]
+    pub struct LoggerConfigDiff {
+        #[serde(flatten)]
+        pub default: default::ConfigDiff,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize, SmartDefault)]
+    #[serde(from = "helpers::SpanEvents", into = "helpers::SpanEvents")]
+    pub struct SpanEvents {
+        #[default(fmt::format::FmtSpan::NONE)]
+        events: fmt::format::FmtSpan,
+    }
+
+    impl From<fmt::format::FmtSpan> for SpanEvents {
+        fn from(events: fmt::format::FmtSpan) -> Self {
+            Self { events }
+        }
+    }
+
+    impl From<SpanEvents> for fmt::format::FmtSpan {
+        fn from(events: SpanEvents) -> Self {
+            events.events
+        }
+    }
+
+    #[derive(Copy, Clone, Debug, Deserialize, Serialize, SmartDefault)]
+    #[serde(from = "helpers::Color", into = "helpers::Color")]
```

Comment:
I'm not sure if this works with our OpenAPI spec. We should give it a check.

Reply:
Good point. :/

Reply:
I guess, we decided to not include `/logger` in the spec.